rfcs/063-catalogue_pipeline_lambdas/README.md

```diff
@@ -0,0 +1,139 @@
+# Catalogue Pipeline services from ECS to Lambda 
+
+## Context
+
+Catalogue-pipeline services run on [Elastic Container Service (ECS)](https://aws.amazon.com/ecs/). The infrastructure includes scaling to account for resource needs during a reindex (scale up) or day-to-day running of the pipeline (scale down). Scaling is achieved using ECS auto scaling rules driven from CloudWatch metrics on queue depth. Other newer projects leverage AWS Lambda to run similar tasks, and there is an opportunity to remove complexity in the catalogue pipeline and align with newer projects.
+
+## Background
+
+The following points cover in more detail the reasoning behind moving from ECS based compute to AWS Lamdba:
+
+- **Reduction in infra complexity by removing ECS auto-scaling logic:**
+
+   Scaling the pipeline up or down is achieved in several ways:
+   
+   - A manual step is performed to terraform the pipeline into "reindexing_state" by setting a bunch of variables to `true` [here](pipeline/terraform/2024-08-15/main.tf). Search for `reindexing_state` in the catalogue-pipeline repo to find all the 
+     ways these variables are used.  
+   - Every service also has a high/low message count alarm on its input queue. By [default](pipeline/terraform/modules/stack/variables.tf) services scale up to 12 instances as long as there are messages on the queue, and will scale down when the queue 
+     is clear. However some services (eg. ingestors) have specially configured `max_capacity` usually to account for data stores' read/write limits.
+
+   This change aims to remove the need for this scaling logic.
+
+- **Reduction in app complexity by removing akka/pekko batching:**
+
+   Several (SQS driven) pipeline services implement “SQSStream” (see https://github.com/wellcomecollection/scala-libs/blob/main/messaging/src/main/scala/weco/messaging/sqs/SQSStream.scala ) that batches SQS messages so that they can be processed in 
+   bulk. This logic lives inside the service and makes it difficult (impossible?) to run locally.  
+
+   This model means the messaging and business/processing logic part of the services are entangled in sometimes confusing ways, and we're losing queue management features of AWS SQS (eg. having to keep track of success and failure to delete or retry 
+   messaging accordingly).  
+ 
+   This change is also an opportunity to reassess and possibly refactor out some error handling that may be too particular, over-optimising for performance at the expense of clarity. 
+
```

Comment:
Yes, but I'm not sure this work is necessary to facilitate that.  The weird batcher error handling could be rewritten even if we left the rest as-is.

Reply:
I'll change this to be less specific: "an opportunity to reassess and possibly reduce incidental complexity".